test/kitchen/drivers/azure-driver.yml

```diff
@@ -0,0 +1,130 @@
+<%
+  ENV['AZURE_LOCATION'] ||= "North Central US"
+  location = ENV['AZURE_LOCATION']
+%>
+
+<%
+  vm_tags = {
+    'dd_agent_testing': 'dd_agent_testing'
+  }
+  if ENV['DD_PIPELINE_ID']
+    vm_tags['pipeline_id'] = ENV['DD_PIPELINE_ID']
+  else
+    vm_tags['pipeline_id'] = 'nopipeline'
+  end
+%>
+
+provisioner:
+  name: chef_solo
+  product_name: chef
+  product_version: 14.12.9
+  install_strategy: always
+
+driver:
+  name: azurerm
+
+driver_config:
+  subscription_id: <%= ENV['AZURE_SUBSCRIPTION_ID'] %>
+  location: <%= location %>
+  <% if ENV['DD_PIPELINE_ID'] %>
+  azure_resource_group_suffix: pl<%= ENV['DD_PIPELINE_ID'] %>
+  <% else %>
+  azure_resource_group_suffix: plnone
+  <% end %>
+
+platforms:
+# Loop through two lists and output a total matrix of all possible platform + chef versions,
+# for both drivers
+
+
+<%
+    # TEST_PLATFORMS syntax is `short_name1,azure_full_qualified_name1|short_name2,azure_full_qualified_name1`
+    # TEST_PLATFORMS syntax is `short_name1,parent vhd folder,parent_vhd_name|...`
+    azure_test_platforms = ENV['TEST_PLATFORMS'].split('|').map { |p| p.split(',') }
+
+    sizes = [
+      "Standard_D1_v2",
+      "Standard_A1_v2",
+    ]
+
+    windows_sizes = [
+      "Standard_D2_v2"
+    ]
+
+    windows_platforms = []
+    sles11_platforms = []
+    sles15_platforms = []
+
+    idx = 0
+    azure_test_platforms.each do |platform|
+    idx += 1
+
+    host = "azure"
+    platform_name = platform[0] + "-#{host}"
+
+    windows = platform_name.include?("win")
+    sles11 = platform_name.include?("sles-11")
+    sles15 = platform_name.include?("sles-15")
+    windows2008 = windows && platform_name.include?("2008")
+
+    if windows
+      windows_platforms << platform_name
+      size = windows_sizes[idx % windows_sizes.length]
+    else
+      if sles15
+        sles15_platforms << platform_name
+      end
+      if sles11
+        sles11_platforms << platform_name
+      end
+      size = sizes[idx % sizes.length]
+    end
+
+    vm_username = ENV['VM_USERNAME'] ? ENV['VM_USERNAME'] : "datadog"
+    vm_password = ENV['SERVER_PASSWORD']
+    vm_hyperv_switch = ENV['KITCHEN_HYPERV_SWITCH'] ? ENV['KITCHEN_HYPERV_SWITCH'] : "public_eth"
```

Comment:
This is repeated in the three driver files and could be in `platforms-common` (except defining `size`, which does change on each iteration and used below), the only change required would be to concatenate `platforms-common` first in the `genconfig` task, before this file, so `vm_username`, `vm_password`... are defined before this loop happens.

Reply:
I don't think you can split it out as you say (unless there's some ruby wizardry that I'm missing).
Those values are used inside some of the driver files (e.g. driver-azure.yml) https://github.com/DataDog/datadog-agent/pull/7457/files#diff-b982e26f120cd92dfaabbfdddcbd01aeb9e9725f46469e639aee16263f2be563R88-R127.
The entire `-name <%= platform_name %>` block is contained in the loop, and requires the platform specific stuff below.